### PR TITLE
ES-795

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    private String policyName;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.expiry}")
+    private String policyExpiry;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.issuedOn}")
+    private String policyIssuedOn;
+
+    @Value("${mosip.digitalcard.sunbird.vci.policy.id}")
+    private String policyId;
```
Comment (vishwa-vyom) [inline src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java:55]: can you rename this as mosip.digitalcard.sunbird.vci.policy.id-prefix and policyIdPrefix ?

File: src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        try {
+            registrySearchRequestDto.setOffset(0);
+            registrySearchRequestDto.setLimit(1);
+            filter.put("eq",policyNumber);
+            filters.put("policyNumber",filter);
+            registrySearchRequestDto.setFilters(filters);
+            List<String> pathsegments = null;
+            String response=restClient.postApi(searchRegistryURL,null,"","", MediaType.APPLICATION_JSON,registrySearchRequestDto,String.class);
+            if(response.contains("osid")){
+                searchPolicy(generatePolicyNumber(policyId));
```
Comment (vishwa-vyom) [inline src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java:91]: This goes into a infinite resursive loop, possible to break it after 15 (config) times ?

File: pom.xml [hunk 3]
```diff
@@ -11,7 +11,7 @@
 	</parent>
 	<groupId>io.mosip.digitalcard</groupId>
 	<artifactId>digital-card-service</artifactId>
-	<version>1.2.0.1-SNAPSHOT</version>
+	<version>1.2.0.1-ES</version>
```
[the remark this answers is not in the document]
Comment (dhanendra06) [reply pom.xml]: taken care, its not be the issue if keep snapshot.

File: src/main/java/io/mosip/digitalcard/service/impl/DigitalCardServiceImpl.java [hunk 4]
```diff
@@ -132,11 +146,15 @@ public void generateDigitalCard(String credential, String credentialType,String
                     throw new DigitalCardServiceException(DigitalCardServiceErrorCodes.DIGITAL_CARD_NOT_GENERATED.getErrorCode(),DigitalCardServiceErrorCodes.DIGITAL_CARD_NOT_GENERATED.getErrorMessage());
                 }
             }
-            if (isPasswordProtected) {
+            pdfCardServiceImpl.generateCard(decryptedCredentialJson, credentialType,password,additionalAttributes);
```
[the remark this answers is not in the document]
Comment (dhanendra06) [reply src/main/java/io/mosip/digitalcard/service/impl/DigitalCardServiceImpl.java]: taken care

File: src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        createInsuranceDto.setDob(convertDateOfBirth((String) identity.get("dateOfBirth")));
+        createInsuranceDto.setEmail((String) identity.get("email"));
+        createInsuranceDto.setGender(formatGender((String) identity.get("gender_eng")));
+        createInsuranceDto.setMobile((String) identity.get("phone"));
+        createInsuranceDto.setBenefits(sunbirdVciBenefits);
+        createInsuranceDto.setPolicyName(policyName);
+        createInsuranceDto.setPolicyExpiresOn(policyExpiry);
+        createInsuranceDto.setPolicyIssuedOn(policyIssuedOn);
+        createInsuranceDto.setPolicyNumber(searchPolicy(generatePolicyNumber(policyId)));
+        createInsuranceDto.setPsut(generatePsut((String) identity.get("RID"),null));
```
[the remark this answers is not in the document]
Comment (dhanendra06) [reply src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java]: taken care

File: src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java (hunk 2 above)
Comment (dhanendra06) [reply src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java]: taken care

Comment (vishwa-vyom) [reply src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java]: This implementation is not correct. inside the while loop we need to generate an id and check again. but we are looping the same result those many times.

Comment (dhanendra06) [reply src/main/java/io/mosip/digitalcard/service/impl/SunbirdVCIServiceImpl.java]: taken care

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**